packages/shell-api/src/helpers.ts

```diff
@@ -295,25 +299,27 @@ export async function getPrintableShardStatus(
     };
 
     if (verbose) {
-      result[mongosAdjective] = await (await mongosColl.find(recentMongosQuery))
+      result[`${mongosAdjective} mongoses`] = await (
+        await mongosColl.find(recentMongosQuery)
+      )
         .sort({ ping: -1 })
         .toArray();
     } else {
-      result[mongosAdjective] = (
+      result[`${mongosAdjective} mongoses`] = (
         (await (
           await mongosColl.aggregate([
             { $match: recentMongosQuery },
             { $group: { _id: '$mongoVersion', num: { $sum: 1 } } },
             { $sort: { num: -1 } },
           ])
-        ).toArray()) as any[]
+        ).toArray()) as { _id: string; num: number }[]
       ).map((z: { _id: string; num: number }) => {
         return { [z._id]: z.num };
       });
     }
   }
 
-  const balancerRes: Record<string, any> = {};
+  const balancerRes = {} as ShardingStatusResult['balancer'];
```

Comment:
Why the preference for it this way around as opposed to:

```
const balancerRes: ShardingStatusResult['balancer'] = {};
```

Reply:
I'm not a fan of this myself but it'd actually be invalid; we're essentially doing a cast here to an invalid object, kind of breaking type safety; still better than not doing a record. Here's a [playground example of what I mean](https://www.typescriptlang.org/play/?#code/PTAEHUFMBsGMHsC2lQBd5oBYoCoE8AHSAZVgCcBLA1UABWgEM8BzM+AVwDsATAGiwoBnUENANQAd0gAjQRVSQAUCEmYKsTKGYUAbpGF4OY0BoadYKdJMoL+gzAzIoz3UNEiPOofEVKVqAHSKymAAmkYI7NCuqGqcANag8ABmIjQUXrFOKBJMggBcISGgoAC0oACCbvCwDKgU8JkY7p7ehCTkVDQS2E6gnPCxGcwmZqDSTgzxxWWVoASMFmgYkAAeRJTInN3ymj4d-jSCeNsMq-wuoPaOltigAKoASgAywhK7SbGQZIIz5VWCFzSeCrZagNYbChbHaxUDcCjJZLfSDbExIAgUdxkUBIursJzCFJtXydajBVDtOYAXlAAG9FCUStx8ldUJROMwANyKAC+wQQnEENAYADVHKAabSeWJhBVuaLHAFXDSAEQAZhV3IFQrEYrIACYWVVJTytY0dQqyGqjRK6czQOqVabgiocM0PGQvIh4H0GMD2DQvm4zMx2AxmJB+LBoOpEn74HoRF4VQBRVYMRALfQqpLYlXgByoADkwgAcpAJCqgioAPJfMjvQSRrSQI6oRwKVzSPCgJzenTDLD6FAIRDQ4SXIMSH3REQGDjYhZMVgcHhBEpAA)

Reply:
Wondering if it is better to only assert once it is correct rather than cast. Just a style nit, really.

Reply:
I think the benefit of the cast is that it gives us IDE hints so at least it adds some checks while the fields are being set; though I agree it also can give misleading idea of what fields are already set depending on context.

Seems fine for initialization though; do you mean like a runtime assertion function?

Reply:
I just mean that at line 322 it isn't really a ShardingStatusResult['balancer']. But I don't have better ideas.